packages/go/analysis/azure/post.go

```diff
@@ -993,26 +1027,63 @@ func ExecuteCommand(ctx context.Context, db graph.Database) (*analysis.AtomicPos
+	return operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
+		if pwResetRoles, err := TenantRoles(tx, tenant, ResetPasswordRoleIDs()...); err != nil {
+			return err
+		} else {
+			for _, role := range pwResetRoles {
+				if roleTemplateIDProp := role.Properties.Get(azure.RoleTemplateID.String()); roleTemplateIDProp.IsNil() {
+					log.Errorf("unable to process azresetpassword for role node %d - missing property %s", role.ID, azure.RoleTemplateID)
+				} else if roleTemplateID, err := roleTemplateIDProp.String(); err != nil {
+					log.Errorf("unable to process azresetpassword for role node %d - property %s is not a string", role.ID, azure.RoleTemplateID)
+				} else {
+					targets := roaring.New()
+					switch roleTemplateID {
+					case constants.GlobalAdministratorRoleID, constants.PrivilegedAuthenticationAdministratorRoleID, constants.PartnerTier2SupportRoleID:
+						targets.Or(roleAssignments.Users())
+					case constants.UserAdministratorRoleID:
+						targets.Or(roleAssignments.UsersWithoutRoles())
+						targets.Or(roleAssignments.UsersWithRolesExclusive(UserAdministratorPasswordResetTargetRoles()...))
+					case constants.HelpdeskAdministratorRoleID:
+						targets.Or(roleAssignments.UsersWithoutRoles())
+						targets.Or(roleAssignments.UsersWithRolesExclusive(HelpdeskAdministratorPasswordResetTargetRoles()...))
+					case constants.AuthenticationAdministratorRoleID:
+						targets.Or(roleAssignments.UsersWithoutRoles())
+						targets.Or(roleAssignments.UsersWithRolesExclusive(AuthenticationAdministratorPasswordResetTargetRoles()...))
+					case constants.PasswordAdministratorRoleID:
+						targets.Or(roleAssignments.UsersWithoutRoles())
+						targets.Or(roleAssignments.UsersWithRolesExclusive(PasswordAdministratorPasswordResetTargetRoles()...))
+					case constants.PartnerTier1SupportRoleID:
+						targets.Or(roleAssignments.UsersWithoutRoles())
+					default:
+						log.Errorf("unable to process azresetpassword for role node %d - encountered unhandled role template id '%s'", role.ID, roleTemplateID)
+					}
+
+					iter := targets.Iterator()
+					for iter.HasNext() {
+						nextJob := analysis.CreatePostRelationshipJob{
+							FromID: role.ID,
+							ToID:   graph.ID(iter.Next()),
+							Kind:   azure.ResetPassword,
+						}
 
-	if securityGroupOwners, err := getRoleEligibleSecurityGroupUsers(ctx, db, roleAssignments); err != nil {
-		return err
-	} else {
-		for _, userID := range roleAssignments.Nodes.AllNodeIDs() {
-			if err := resetPasswordCases(roleAssignments, operation, userID, usersWithoutRoles, securityGroupOwners); err != nil {
-				log.Errorf("Unable to process AZResetPassword for node %d: %v", userID, err)
+						if !channels.Submit(ctx, outC, nextJob) {
+							return nil
+						}
+
+					}
+				}
```

Comment:
This entire pipeline should probably be broken out into a separate function to help reduce nesting and make the logic a little easier to track